autoreduce_qp/queue_processor/reduction/process_manager.py

```diff
@@ -32,75 +31,67 @@ def __init__(self, message: Message, run_name: str, software: Software) -> None:
-                # The environment variables used are the same as those used by the Docker command-line client
-                # https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
-                client = docker.from_env()
-
-                image = get_correct_image(client, self.software)
-
-                if "AUTOREDUCTION_PRODUCTION" in os.environ:
-                    reduced_data = Path('/instrument')
-                else:
-                    reduced_data = Path(f'{PROJECT_DEV_ROOT}/reduced-data')
-                    if not os.path.exists(ARCHIVE_ROOT):
-                        Path(ARCHIVE_ROOT).mkdir(parents=True, exist_ok=True)
-                    if not os.path.exists(reduced_data):
-                        reduced_data.mkdir(parents=True, exist_ok=True)
-
-                    # Run chmod's to make sure the directories are writable
-                    reduced_data.chmod(0o777)
-                    Path(ARCHIVE_ROOT).chmod(0o777)
-                    Path(f'{AUTOREDUCE_HOME_ROOT}/logs/autoreduce.log').chmod(0o777)
-
-                container = client.containers.run(
-                    image=image,
-                    command=args,
-                    volumes={
-                        AUTOREDUCE_HOME_ROOT: {
-                            'bind': '/home/isisautoreduce/.autoreduce/',
-                            'mode': 'rw'
-                        },
-                        ARCHIVE_ROOT: {
-                            'bind': '/isis/',
-                            'mode': 'rw'
-                        },
-                        reduced_data: {
-                            'bind': '/instrument/',
-                            'mode': 'rw'
-                        },
-                        temp_dir: {
-                            'bind': '/output/',
-                            'mode': 'rw'
-                        },
+            # We need to run the reduction in a new process, otherwise scripts
+            # will fail when they use things that require access to a main loop
+            # e.g. a GUI main loop, for matplotlib or Mantid
+            serialized_vars = self.message.serialize()
+            serialized_vars_truncated = self.message.serialize(limit_reduction_script=True)
+            args = ["autoreduce-runner-start", serialized_vars, self.run_name]
+            logger.info("Calling: %s %s %s %s ", "python3", "runner.py", serialized_vars_truncated, self.run_name)
+
+            # Return a client configured from environment variables
+            # The environment variables used are the same as those used by the Docker command-line client
+            # https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
+            client = docker.from_env()
+
+            image = get_correct_image(client, self.software)
+
+            if "AUTOREDUCTION_PRODUCTION" in os.environ:
+                reduced_data = Path('/instrument')
```

Comment:
Could this line be added to our test code coverage?

Reply:
I don't think it's worth it as it's just a variable assignment. Codecov just has a weird thing with if-else statements (even if it's just assigning a variable)

Reply:
I appreciate that, however, the code after this variable assignment has different output based on the variable assignment right?